### PR TITLE
Really Bail out in Base::is_success

### PR DESCRIPTION
**Pull request by szepeviktor**
@JJJ You bail out everywhere except here.

```php
		// Bail if no row exists
		if ( empty( $result ) ) {
			return false;

		// Bail if an error occurred
		if ( is_wp_error( $result ) ) {
			$this->last_error = $result;
			return false;
		}

		// No errors
		return true;
```

### CODE REVIEW

File: src/Database/Base.php [hunk 1]
```diff
@@ -144,9 +144,9 @@ public function to_array() {
 	 * @return string
 	 */
 	protected function apply_prefix( $string = '', $sep = '_' ) {
-		return ! empty( $this->prefix )
-			? "{$this->prefix}{$sep}{$string}"
-			: $string;
+		return empty( $this->prefix )
+			? $string
+			: "{$this->prefix}{$sep}{$string}";
```
Comment (szepeviktor) [inline src/Database/Base.php:149]: Spare one exclamation mark here.

Comment (szepeviktor) [reply src/Database/Base.php:149]: Actually it should be

```php
 		if empty( $this->prefix ) {
			return $string;
		}

		return "{$this->prefix}{$sep}{$string}"; 
```

[kép](https://gist.github.com/szepeviktor/9b70cab1b3313ff35e0ac1ace2d63e62)

Comment (JJJ) [reply src/Database/Base.php:149]: The release/2.1.0 branch has somewhat inadvertently resolved this 👍😅

File: src/Database/Base.php [hunk 2]
```diff
@@ -325,19 +325,15 @@ protected function is_success( $result = false ) {
...
-		// No errors
-		} else {
-			$retval = true;
+			return false;
 		}
 
-		// Return the result
-		return (bool) $retval;
+		// No errors
+		return true;
```
Comment (JJJ) [inline src/Database/Base.php:337]: I don't think it is correct for this to assume it was successful.

It should default to fail, and only succeed if non-empty & non-erroneous.

Kind of like checking if a password is correct?

**Comment by JJJ** [comment]
Updating in issue/128 PR - thank you @szepeviktor I appreciate it

**Pull request closed without merging**